test/layeredmaterialnodeprocessing_unit_test.js

```diff
@@ -75,6 +78,8 @@ describe('updateLayeredMaterialNodeImagery', function () {
         tile.material.indexOfColorLayer = () => 0;
         tile.material.isColorLayerDownscaled = () => true;
         tile.material.getColorLayerLevelById = () => 1;
+
+        updateLayeredMaterialNodeImagery(context, layer, tile);
```

Comment:
If there needs to be a second call, you should add a test to validate the intermediate state.

I suppose that you want to check that the first didn't make any network fetch, since it's the feature added by your PR.

Reply:
done